kotlinx-coroutines-core/common/src/channels/ConflatedChannel.kt

```diff
@@ -15,80 +12,116 @@ import kotlinx.coroutines.internal.*
  * Sender to this channel never suspends and [offer] always returns `true`.
  *
  * This channel is created by `Channel(Channel.CONFLATED)` factory function invocation.
- *
- * This implementation is fully lock-free.
  */
 internal open class ConflatedChannel<E> : AbstractChannel<E>() {
-    protected final override val isBufferAlwaysEmpty: Boolean get() = true
-    protected final override val isBufferEmpty: Boolean get() = true
-    protected final override val isBufferAlwaysFull: Boolean get() = false
-    protected final override val isBufferFull: Boolean get() = false
+    protected final override val isBufferAlwaysEmpty: Boolean = false
+    protected final override val isBufferEmpty: Boolean get() = lock.withLock { size == 0 }
+    protected final override val isBufferAlwaysFull: Boolean = false
+    protected final override val isBufferFull: Boolean get() = lock.withLock { size == 1 }
```

Comment:
This override of `isBufferFull` should not be needed. Conceptually, a conflated channel is never full, since you can also successfully send to it without suspension. Correspondingly, the override of `isFullImpl` should not be needed either.